raiden/tests/utils/smoketest.py

```diff
@@ -373,104 +316,91 @@ def setup_raiden(transport, matrix_server, print_step, contracts_version, testch
-            raiden_api = RaidenAPI(app.raiden)
-            rest_api = RestAPI(raiden_api)
-            (api_host, api_port) = split_endpoint(args["api_address"])
-            api_server = APIServer(rest_api, config={"host": api_host, "port": api_port})
-            api_server.start()
-
-            block = app.raiden.get_block_number() + DEFAULT_NUMBER_OF_BLOCK_CONFIRMATIONS
-            # Proxies now use the confirmed block hash to query the chain for
-            # prerequisite checks. Wait a bit here to make sure that the confirmed
-            # block hash contains the deployed token network or else things break
-            wait_for_block(raiden=app.raiden, block_number=block, retry_timeout=1.0)
-
-            raiden_api.channel_open(
-                registry_address=contract_addresses[CONTRACT_TOKEN_NETWORK_REGISTRY],
-                token_address=to_canonical_address(token.contract.address),
-                partner_address=to_canonical_address(TEST_PARTNER_ADDRESS),
-            )
-            raiden_api.set_total_channel_deposit(
-                contract_addresses[CONTRACT_TOKEN_NETWORK_REGISTRY],
-                to_canonical_address(token.contract.address),
-                to_canonical_address(TEST_PARTNER_ADDRESS),
-                TEST_DEPOSIT_AMOUNT,
-            )
-            token_addresses = [to_checksum_address(token.contract.address)]
-
-            print_step("Running smoketest")
-            error = smoketest_perform_tests(app.raiden, token_addresses)
-            if error is not None:
-                append_report("Smoketest assertion error", error)
-            else:
-                success = True
-        except:  # noqa pylint: disable=bare-except
-            if debug:
-                import pdb
-
-                # The pylint comment is required when pdbpp is installed
-                pdb.post_mortem()  # pylint: disable=no-member
-            else:
-                error = traceback.format_exc()
-                append_report("Smoketest execution error", error)
-        finally:
-            if app is not None:
-                app.stop()
-                app.raiden.get()
-            node_executor = ethereum_nodes[0]
-            node = node_executor.process
-            node.send_signal(signal.SIGINT)
-            try:
-                node.wait(10)
-            except TimeoutExpired:
-                print_step("Ethereum node shutdown unclean, check log!", error=True)
-                node.kill()
-
-            if isinstance(node_executor.stdio, tuple):
-                logfile = node_executor.stdio[1]
-                logfile.flush()
-                logfile.seek(0)
-                append_report("Ethereum Node log output", logfile.read())
-    append_report("Raiden Node stdout", raiden_stdout.getvalue())
-    if success:
-        print_step(f"Smoketest successful")
-    else:
-        print_step(f"Smoketest had errors", error=True)
-    return success
+    app = None
+    api_server = None
```

Comment:
`api_server` was not being `stop`ed and `get`ed, so errors there could be missed by the smoketest.